src/tc-yaml.js

```diff
@@ -252,19 +252,24 @@ class VersionOne extends TcYaml {
       }
 
       config.tasks = config.tasks.map(task => {
-        task = _.defaults(task, {
+        task = Object.assign({
           taskId: defaultTaskId,
           taskGroupId: defaultTaskGroupId,
           created: now,
-        });
+          routes: config.reporting ? [cfg.app.checkTaskRoute] : [cfg.app.statusTaskRoute],
+        }, task);
         defaultTaskId = slugid.nice(); // invent a new taskId for the next task
 
-        return {
-          taskId: task.taskId,
-          task: _.omit(
-            _.extend(task, {schedulerId: cfg.taskcluster.schedulerId}),
-            'taskId'),
+        const {taskId, ...taskWithoutTaskId} = task;
+        const b = {
+          taskId,
+          task: {
+            ...taskWithoutTaskId,
+            schedulerId: cfg.taskcluster.schedulerId,
```

Comment:
Why not add `schedulerId` first and then just return `{taskId, ...taskWithoutTaskId}`?

Reply:
I'm not sure I see the difference between the two options

Reply:
oh, `const b` is a debugging left over! I'll clean it up, it shouldn't be there